src/Modal.js

```diff
@@ -155,10 +155,13 @@ class Modal extends React.Component {
       this._element = null;
     }
 
-    // Use regex to prevent matching `modal-open` as part of a different class, e.g. `my-modal-opened`
-    const classes = document.body.className.replace(/(^| )modal-open( |$)/, ' ');
-    document.body.className = mapToCssModules(classNames(classes).trim(), this.props.cssModule);
-    setScrollbarWidth(this.originalBodyPadding);
+    if (!this.props.isOpen) {
```

Comment:
Not a good check. In case that Modal never wasn't open before, it's still will removing class from `body`.

Reply:
Good catch. I ended up just resorting to keeping track of whether or not the instance added or removed the class. This way it will only ever add it if it isn't there and only remove it if it was added.